From 5849a4a989005b62f5ef0be22267f2c72027b21d Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 29 Nov 2016 21:45:48 +0800
Subject: [PATCH 1297/5242] MLK-13534-1 ARM: dts: imx7ulp: add usdhc1 node

commit  0d4d828b35603124e613deab2adced4fbb503797 from
https://source.codeaurora.org/external/imx/linux-imx.git

There are 2 usdhc instances on i.MX7ULP, previous
usdhc1 should be usdhc0, now add the correct usdhc1 node.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk-emmc.dts |    6 +++---
 arch/arm/boot/dts/imx7ulp-evk.dts      |   10 +++++-----
 arch/arm/boot/dts/imx7ulp.dtsi         |   14 +++++++++++++-
 3 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk-emmc.dts b/arch/arm/boot/dts/imx7ulp-evk-emmc.dts
index 4f667b8..e58616e 100644
--- a/arch/arm/boot/dts/imx7ulp-evk-emmc.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk-emmc.dts
@@ -8,10 +8,10 @@
 
 #include "imx7ulp-evk.dts"
 
-&usdhc1 {
+&usdhc0 {
 	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&pinctrl_usdhc1_8bit>;
-	pinctrl-1 = <&pinctrl_usdhc1_8bit>;
+	pinctrl-0 = <&pinctrl_usdhc0_8bit>;
+	pinctrl-1 = <&pinctrl_usdhc0_8bit>;
 	non-removable;
 	bus-width = <8>;
 	status = "okay";
diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 710bc35..cdd25ba 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -37,7 +37,7 @@
 			>;
 		};
 
-		pinctrl_usdhc1: usdhc1grp {
+		pinctrl_usdhc0: usdhc0grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTD1__SDHC0_CMD	0x843
 				IMX7ULP_PAD_PTD2__SDHC0_CLK	0x843
@@ -48,7 +48,7 @@
 			>;
 		};
 
-		pinctrl_usdhc1_8bit: usdhc1grp_8bit {
+		pinctrl_usdhc0_8bit: usdhc0grp_8bit {
 			fsl,pins = <
 				IMX7ULP_PAD_PTD1__SDHC0_CMD	0x843
 				IMX7ULP_PAD_PTD2__SDHC0_CLK	0x843
@@ -101,10 +101,10 @@
 	};
 };
 
-&usdhc1 {
+&usdhc0 {
 	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&pinctrl_usdhc1>;
-	pinctrl-1 = <&pinctrl_usdhc1>;
+	pinctrl-0 = <&pinctrl_usdhc0>;
+	pinctrl-1 = <&pinctrl_usdhc0>;
 	non-removable;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 39526a1..840295b 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -292,7 +292,7 @@
 			nxp,sim = <&sim>;
 		};
 
-		usdhc1: usdhc@40370000 {
+		usdhc0: usdhc@40370000 {
 			compatible = "fsl,imx7ulp-usdhc";
 			reg = <0x40370000 0x10000>;
 			interrupts = <GIC_SPI 42 IRQ_TYPE_LEVEL_HIGH>;
@@ -304,6 +304,18 @@
 			status = "disabled";
 		};
 
+		usdhc1: usdhc@40380000 {
+			compatible = "fsl,imx7ulp-usdhc";
+			reg = <0x40380000 0x10000>;
+			interrupts = <GIC_SPI 43 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clks IMX7ULP_CLK_NIC1_BUS_DIV>,
+				 <&clks IMX7ULP_CLK_NIC1_DIV>,
+				 <&clks IMX7ULP_CLK_USDHC1>;
+			clock-names ="ipg", "ahb", "per";
+			bus-width = <4>;
+			status = "disabled";
+		};
+
 		wdog1: wdog@403D0000 {
 			compatible = "fsl,imx7ulp-wdt";
 			reg = <0x403D0000 0x10000>;
-- 
1.7.9.5

